example/t3dn_bip_example/ops.py

```diff
@@ -110,18 +112,107 @@ def execute(self, context: bpy.types.Context) -> set:
         return {'FINISHED'}
 
 
+icon_items: List[Tuple[str, str, str]] = []
+
+
+def get_icon_items(self, context: bpy.types.Context):
+    paths = previews.folder.joinpath(self.type).glob(f'*.{self.type}')
+    coll = previews.PREVIEW_COLL['images']
+
+    icon_items.clear()
+
+    for index, path in enumerate(list(paths)):
+        try:
+            icon = coll[str(path)]
+        except KeyError:
+            icon = coll.load(str(path), str(path), 'IMAGE')
```

Comment:
there is a load_safe function which does exactly that.

Reply:
I've replaced it.